pkg/asset/installconfig/installconfig.go

```diff
@@ -133,6 +135,29 @@ func (a *InstallConfig) finishGCP() error {
 		}
 		a.Config.Platform.GCP.Endpoint.ClusterUseOnly = &defaultClusterUseOnly
 	}
+
+	project := a.Config.GCP.ProjectID
+	if a.Config.Platform.GCP.NetworkProjectID != "" {
+		project = a.Config.GCP.NetworkProjectID
+	}
+
+	if a.Config.GCP.FirewallRulesManagement == "" {
+		firewallPermissions, err := icgcp.HasPermission(context.TODO(), project, []string{
+			icgcp.CreateFirewallPermission,
+			icgcp.DeleteFirewallPermission,
+			icgcp.UpdateNetworksPermission,
+		}, a.Config.GCP.Endpoint)
+		if err != nil {
+			return err
+		}
+
+		a.Config.GCP.FirewallRulesManagement = gcp.ManagedFirewallRules
+		if !firewallPermissions && a.Config.Platform.GCP.NetworkProjectID != "" {
```

Comment:
And for the case where `!firewallPermissions && a.Config.Platform.GCP.NetworkProjectID == ""` we should have an error indicating the permissions are insufficient, so the user should either add the permissions or set Unmanaged and provide their own firewalls.

Reply:
Hm but ideally that validation should be done if FirewallRulesManagement is unset OR set to managed, so it might not fit perfectly in this code block.